File: app/src/main/java/uk/co/agfsoft/skyweather/model/WeatherCity.java
package uk.co.agfsoft.skyweather.model;


import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;



public class WeatherCity extends RealmObject {
    @PrimaryKey
    int id;
    String name;
    String country;
    double latitiude;
    double longitude;

    public WeatherCity() {
    }

    public WeatherCity(RawWeatherCity rawWeatherCity) {
        this.id = rawWeatherCity.getId();
        this.name = rawWeatherCity.getName();
        this.country = rawWeatherCity.getCountry();
        this.longitude = rawWeatherCity.getLongitude();
        this.latitiude = rawWeatherCity.getLatitude();
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getCountry() {
        return country;
    }

    public double getLatitiude() {
        return latitiude;
    }

    public double getLongitude() {
        return longitude;
    }


}

File: app/src/main/kotlin/uk/co/agfsoft/skyweather/model/WeatherCity.kt
package model

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey


open class WeatherCity : RealmObject {
    @PrimaryKey
    var id: Int = 0
        internal set
    var name: String? = null
        internal set
    var country: String? = null
        internal set
    var latitiude: Double = 0.toDouble()
        internal set
    var longitude: Double = 0.toDouble()
        internal set

    constructor() {}

    constructor(rawWeatherCity: RawWeatherCity) {
        this.id = rawWeatherCity.id
        this.name = rawWeatherCity.name
        this.country = rawWeatherCity.country
        this.longitude = rawWeatherCity.longitude
        this.latitiude = rawWeatherCity.latitude
    }


}
